src/tribler-common/tribler_common/network_utils.py

```diff
@@ -18,6 +18,35 @@ class FreePortNotFoundError(Exception):
     pass
 
 
+class NetworkUtils:
+    def __init__(self, socket_class=None):
+        if socket_class is None:
+            socket_class = lambda: socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+        self.socket_class = socket_class
+
+    def get_first_free_port(self, start=5000, limit=100):
+        stop = min(MAX_PORT, start + limit)
+        logger.info(f'Looking for first free port in range [{start}..{stop}]')
+
+        for port in range(start, stop):
+            if self.is_port_free(port):
+                logger.info(f'{port} is free')
+                return port
+
+            logger.info(f'{port} in use')
+
+        raise FreePortNotFoundError(f'Free port not found in range [{start}..{stop}]')
+
+    def is_port_free(self, port):
+        try:
+            with self.socket_class() as s:
+                s.bind(('', port))
+                return True
+        except OSError:
+            return False
+
+
 def get_first_free_port(start=5000, limit=100, family=socket.AF_INET, socket_type=socket.SOCK_STREAM):
```

Comment:
You have refactored this function so nicely above, why didn't you remove the original?

Reply:
I did it to keep an illustration of the power of refactoring :) (joke)

I've just forgotten to commit this. Thank you.